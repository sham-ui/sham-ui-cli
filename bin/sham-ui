#!/usr/bin/env node

require( 'commander' )
    .version( require( '../package' ).version )
    .usage( '<command> [options]' )
    .command( 'new', 'generate a new project from a template' ).alias( 'n' )
    .command( 'upgrade', 'upgrade existed project from a template' ).alias( 'u' )
    .command( 'generate', 'generate a new component' ).alias( 'g' )
    .command( 'collect', 'collect filters & directives' ).alias( 'c' )
    .command( 'destroy', 'destroy component' ).alias( 'd' )
    .parse( process.argv );