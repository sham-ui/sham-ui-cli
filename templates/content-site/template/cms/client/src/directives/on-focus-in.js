import { EventListener } from 'sham-ui-directives';

export class onFocusIn extends EventListener {
    constructor() {
        super( 'focusin' );
    }
}
