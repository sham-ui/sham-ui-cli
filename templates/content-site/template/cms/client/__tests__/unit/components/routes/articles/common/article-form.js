import { createDI } from 'sham-ui';
import * as directives from 'sham-ui-directives';
import renderer from 'sham-ui-test-helpers';
import { onFocusIn } from '../../../../../../src/directives/on-focus-in';
// eslint-disable-next-line max-len
import RoutesArticlesCommonArticleForm from '../../../../../../src/components/routes/articles/common/article-form.sfc';

it( 'renders correctly', async() => {
    const DI = createDI();
    DI.bind( 'store', {
        articleCategories: jest.fn().mockReturnValueOnce(
            Promise.resolve( {
                categories: [
                    { 'id': '4', 'name': 'Hello world!', 'slug': 'hello-world' },
                    { 'id': '6', 'name': 'Название категории', 'slug': 'nazvanie-kategorii' }
                ],
                meta: {}
            } )
        ),
        articleTags: jest.fn().mockReturnValueOnce(
            Promise.resolve( {
                tags: [
                    { 'id': '4', 'name': 'Hello world!', 'slug': 'hello-world' },
                    { 'id': '6', 'name': 'Название тега', 'slug': 'nazvanie-tega' }
                ],
                meta: {}
            } )
        )
    } );
    const meta = renderer( RoutesArticlesCommonArticleForm, {
        DI,
        directives: {
            onFocusIn,
            ...directives
        },
        publishedAt: new Date( '2022-04-29T12:24:51.637Z' )
    } );
    await new Promise( resolve => setImmediate( resolve ) );
    expect( meta.toJSON() ).toMatchSnapshot();
} );


it( 'fail load data', async() => {
    const DI = createDI();
    DI.bind( 'store', {
        articleCategories: jest.fn().mockReturnValueOnce(
            Promise.resolve( {
                categories: [
                    { 'id': '4', 'name': 'Hello world!', 'slug': 'hello-world' },
                    { 'id': '6', 'name': 'Название категории', 'slug': 'nazvanie-kategorii' }
                ],
                meta: {}
            } )
        ),
        articleTags: jest.fn().mockReturnValueOnce(
            Promise.reject( {} )
        )
    } );
    const meta = renderer( RoutesArticlesCommonArticleForm, {
        DI,
        directives: {
            onFocusIn,
            ...directives
        },
        publishedAt: new Date( '2022-04-29T12:24:51.637Z' )
    } );
    await new Promise( resolve => setImmediate( resolve ) );
    expect( meta.toJSON() ).toMatchSnapshot();
} );
