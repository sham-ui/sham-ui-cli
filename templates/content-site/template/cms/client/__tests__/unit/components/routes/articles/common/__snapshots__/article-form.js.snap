// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fail load data 1`] = `
Object {
  "Options": Object {
    "allTags": Array [],
    "body": "",
    "categories": Array [
      Object {
        "id": "4",
        "name": "Hello world!",
        "slug": "hello-world",
      },
      Object {
        "id": "6",
        "name": "Название категории",
        "slug": "nazvanie-kategorii",
      },
    ],
    "category": null,
    "dataLoaded": true,
    "errors": Array [
      "Fail load data",
    ],
    "onSaveSuccess": [Function],
    "publishedAt": 2022-04-29T12:24:51.637Z,
    "save": [Function],
    "shortBody": "",
    "tags": Array [],
    "title": "",
  },
  "html": "
<ul class=\\"errors-panel\\">
  <li>Fail load data</li>
</ul>
<!--0-->
<!--0-->
<form class=\\"form-layout\\">
  <!--0-->
  <!--0-->
  <!--1-->
  <div class=\\"form-group\\" data-test-field-title=\\"\\">
    <label>Title:</label>
    <input autofocus=\\"\\">
  </div>
  <div class=\\"form-group\\" data-test-field-category=\\"\\">
    <label>Category:</label>
    <select>
      <option value=\\"null\\">- - -</option>
      <option value=\\"4\\">Hello world!</option>
      <option value=\\"6\\">Название категории</option>
    </select>
    <!--0-->
  </div>
  <div class=\\"form-group\\" data-test-field-short-body=\\"\\">
    <label>Short body:</label>
    <textarea></textarea>
  </div>
  <div class=\\"form-group\\" data-test-field-body=\\"\\">
    <label>Body:</label>
    <textarea></textarea>
    <!--1-->
  </div>
  <div class=\\"form-group\\" data-test-field-tags=\\"\\">
    <label>Tags:</label>
    <div class=\\"tags-input\\">
      <!--0-->
      <div class=\\"autocomplete-input\\">
        <input autocomplete=\\"off\\">
        <!--0-->
      </div>
      <!--1-->
    </div>
    <!--2-->
  </div>
  <div class=\\"form-group\\" data-test-field-published-at=\\"\\">
    <label>Published at:</label>
    <table class=\\"sham-ui-datetimepicker\\">
      <thead>
        <tr>
          <th><span>◂</span></th>
          <th colspan=\\"5\\" class=\\"picker-switch\\">Апрель 2022 г.</th>
          <th><span>▸</span></th>
        </tr>
        <tr>
          <th>Пн</th>
          <th>Вт</th>
          <th>Ср</th>
          <th>Чт</th>
          <th>Пт</th>
          <th>Сб</th>
          <th>Вс</th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td class=\\"day other-month\\">28</td>
          <td class=\\"day other-month\\">29</td>
          <td class=\\"day other-month\\">30</td>
          <td class=\\"day other-month\\">31</td>
          <td class=\\"day\\">1</td>
          <td class=\\"day\\">2</td>
          <td class=\\"day\\">3</td>
        </tr>
        <tr>
          <td class=\\"day\\">4</td>
          <td class=\\"day\\">5</td>
          <td class=\\"day\\">6</td>
          <td class=\\"day\\">7</td>
          <td class=\\"day\\">8</td>
          <td class=\\"day\\">9</td>
          <td class=\\"day\\">10</td>
        </tr>
        <tr>
          <td class=\\"day\\">11</td>
          <td class=\\"day\\">12</td>
          <td class=\\"day\\">13</td>
          <td class=\\"day\\">14</td>
          <td class=\\"day\\">15</td>
          <td class=\\"day\\">16</td>
          <td class=\\"day\\">17</td>
        </tr>
        <tr>
          <td class=\\"day\\">18</td>
          <td class=\\"day\\">19</td>
          <td class=\\"day\\">20</td>
          <td class=\\"day\\">21</td>
          <td class=\\"day\\">22</td>
          <td class=\\"day\\">23</td>
          <td class=\\"day\\">24</td>
        </tr>
        <tr>
          <td class=\\"day\\">25</td>
          <td class=\\"day\\">26</td>
          <td class=\\"day\\">27</td>
          <td class=\\"day\\">28</td>
          <td class=\\"day active\\">29</td>
          <td class=\\"day\\">30</td>
          <td class=\\"day other-month\\">1</td>
        </tr>
        <tr>
          <td class=\\"day other-month\\">2</td>
          <td class=\\"day other-month\\">3</td>
          <td class=\\"day other-month\\">4</td>
          <td class=\\"day other-month\\">5</td>
          <td class=\\"day other-month\\">6</td>
          <td class=\\"day other-month\\">7</td>
          <td class=\\"day other-month\\">8</td>
        </tr>
      </tbody>
      <tfoot>
        <tr>
          <th colspan=\\"7\\">19:24:51</th>
        </tr>
      </tfoot>
      <!--0-->
      <!--0-->
    </table>
    <!--3-->
  </div>
  <!--0-->
  <!--2-->
  <button type=\\"submit\\">
    <!--3-->Save
    <!--4-->
  </button>
  <!--5-->
</form>
<!--0-->
<!--1-->
<!--1-->
<!--0-->
",
}
`;

exports[`renders correctly 1`] = `
Object {
  "Options": Object {
    "allTags": Array [
      Object {
        "id": "4",
        "name": "Hello world!",
        "slug": "hello-world",
      },
      Object {
        "id": "6",
        "name": "Название тега",
        "slug": "nazvanie-tega",
      },
    ],
    "body": "",
    "categories": Array [
      Object {
        "id": "4",
        "name": "Hello world!",
        "slug": "hello-world",
      },
      Object {
        "id": "6",
        "name": "Название категории",
        "slug": "nazvanie-kategorii",
      },
    ],
    "category": null,
    "dataLoaded": true,
    "errors": Array [],
    "onSaveSuccess": [Function],
    "publishedAt": 2022-04-29T12:24:51.637Z,
    "save": [Function],
    "shortBody": "",
    "tags": Array [],
    "title": "",
  },
  "html": "
<!--0-->
<!--0-->
<form class=\\"form-layout\\">
  <!--0-->
  <!--0-->
  <!--1-->
  <div class=\\"form-group\\" data-test-field-title=\\"\\">
    <label>Title:</label>
    <input autofocus=\\"\\">
  </div>
  <div class=\\"form-group\\" data-test-field-category=\\"\\">
    <label>Category:</label>
    <select>
      <option value=\\"null\\">- - -</option>
      <option value=\\"4\\">Hello world!</option>
      <option value=\\"6\\">Название категории</option>
    </select>
    <!--0-->
  </div>
  <div class=\\"form-group\\" data-test-field-short-body=\\"\\">
    <label>Short body:</label>
    <textarea></textarea>
  </div>
  <div class=\\"form-group\\" data-test-field-body=\\"\\">
    <label>Body:</label>
    <textarea></textarea>
    <!--1-->
  </div>
  <div class=\\"form-group\\" data-test-field-tags=\\"\\">
    <label>Tags:</label>
    <div class=\\"tags-input\\">
      <!--0-->
      <div class=\\"autocomplete-input\\">
        <input autocomplete=\\"off\\">
        <!--0-->
      </div>
      <!--1-->
    </div>
    <!--2-->
  </div>
  <div class=\\"form-group\\" data-test-field-published-at=\\"\\">
    <label>Published at:</label>
    <table class=\\"sham-ui-datetimepicker\\">
      <thead>
        <tr>
          <th><span>◂</span></th>
          <th colspan=\\"5\\" class=\\"picker-switch\\">Апрель 2022 г.</th>
          <th><span>▸</span></th>
        </tr>
        <tr>
          <th>Пн</th>
          <th>Вт</th>
          <th>Ср</th>
          <th>Чт</th>
          <th>Пт</th>
          <th>Сб</th>
          <th>Вс</th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td class=\\"day other-month\\">28</td>
          <td class=\\"day other-month\\">29</td>
          <td class=\\"day other-month\\">30</td>
          <td class=\\"day other-month\\">31</td>
          <td class=\\"day\\">1</td>
          <td class=\\"day\\">2</td>
          <td class=\\"day\\">3</td>
        </tr>
        <tr>
          <td class=\\"day\\">4</td>
          <td class=\\"day\\">5</td>
          <td class=\\"day\\">6</td>
          <td class=\\"day\\">7</td>
          <td class=\\"day\\">8</td>
          <td class=\\"day\\">9</td>
          <td class=\\"day\\">10</td>
        </tr>
        <tr>
          <td class=\\"day\\">11</td>
          <td class=\\"day\\">12</td>
          <td class=\\"day\\">13</td>
          <td class=\\"day\\">14</td>
          <td class=\\"day\\">15</td>
          <td class=\\"day\\">16</td>
          <td class=\\"day\\">17</td>
        </tr>
        <tr>
          <td class=\\"day\\">18</td>
          <td class=\\"day\\">19</td>
          <td class=\\"day\\">20</td>
          <td class=\\"day\\">21</td>
          <td class=\\"day\\">22</td>
          <td class=\\"day\\">23</td>
          <td class=\\"day\\">24</td>
        </tr>
        <tr>
          <td class=\\"day\\">25</td>
          <td class=\\"day\\">26</td>
          <td class=\\"day\\">27</td>
          <td class=\\"day\\">28</td>
          <td class=\\"day active\\">29</td>
          <td class=\\"day\\">30</td>
          <td class=\\"day other-month\\">1</td>
        </tr>
        <tr>
          <td class=\\"day other-month\\">2</td>
          <td class=\\"day other-month\\">3</td>
          <td class=\\"day other-month\\">4</td>
          <td class=\\"day other-month\\">5</td>
          <td class=\\"day other-month\\">6</td>
          <td class=\\"day other-month\\">7</td>
          <td class=\\"day other-month\\">8</td>
        </tr>
      </tbody>
      <tfoot>
        <tr>
          <th colspan=\\"7\\">19:24:51</th>
        </tr>
      </tfoot>
      <!--0-->
      <!--0-->
    </table>
    <!--3-->
  </div>
  <!--0-->
  <!--2-->
  <button type=\\"submit\\">
    <!--3-->Save
    <!--4-->
  </button>
  <!--5-->
</form>
<!--0-->
<!--1-->
<!--1-->
<!--0-->
",
}
`;
